tests/test_distributions.py

```diff
@@ -333,6 +498,25 @@ def test_eq_ne_hash() -> None:
         assert not d != d_copy
         assert hash(d) == hash(d_copy)
 
+    # Different field values.
+    di0 = distributions.FloatDistribution(low=1, high=2)
+    di1 = distributions.FloatDistribution(low=1, high=3)
+    assert di0 != di1
+    assert not di0 == di1
+    assert hash(di0) != hash(di1)
```

Comment:
I think `di0 != di1` is sufficient to check equality of distributions because `hash()` encode a class name and parameters, which is checked in `__eq__`.

https://github.com/optuna/optuna/blob/d30263dc47f30015412dd76b8481d37a04cc70ee/optuna/distributions.py#L78-L84

https://github.com/optuna/optuna/blob/d30263dc47f30015412dd76b8481d37a04cc70ee/optuna/distributions.py#L86-L88

Reply:
Agreed, let me remove it.